Projects/Server/Network/RingPipe.cs

```diff
@@ -0,0 +1,271 @@
+using System;
+using System.Runtime.InteropServices;
+
+namespace Server.Network;
+
+public partial class RingPipe : IDisposable
+{
+    public class PipeWriter
+    {
+        private readonly RingPipe _pipe;
+
+        internal PipeWriter(RingPipe pipe) => _pipe = pipe;
+
+        public unsafe Span<byte> TryGetMemory()
+        {
+            var read = _pipe._readIdx;
+            var write = _pipe._writeIdx;
+
+            uint sz;
```

Comment:
Haven't seen anything named sz for a while 😄 it used to be a naming convention for a null terminated string of characters.... back when we put the types of variables in their name 🙂

Does it have a different meaning here?

Reply:
It's short for `size` when `size` might already be used. 🙈